central/sensor/augmentation/broker.go

```diff
@@ -0,0 +1,88 @@
+package augmentation
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/sensor/service/connection"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// The Broker coordinates and matches deployment enhancement requests to responses
+type Broker struct {
+	requests map[string]chan<- *central.DeploymentEnhancementResponse
+	lock     sync.Mutex
+	once     sync.Once
+}
+
+// NewBroker returns a new broker
+func NewBroker() *Broker {
+	return &Broker{
+		requests: make(map[string]chan<- *central.DeploymentEnhancementResponse),
+	}
+}
+
+// NotifyDeploymentReceived matches the ID of Sensors response to the request and notifies the waiting goroutine
+func (b *Broker) NotifyDeploymentReceived(msg *central.DeploymentEnhancementResponse) {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	if msg.GetMsg() == nil {
+		log.Warnf("Received empty message, skipping augmentation notify")
+		return
+	}
+	if r, ok := b.requests[msg.GetMsg().GetId()]; ok {
+		// Once, to prevent writing to a closed channel if a msg dupe arrives
+		b.once.Do(func() {
```

Comment:
I don't think this has the effect your expecting... `b.once` is a `sync.Once` struct in `broker` that will execute this callback *exactly once*. But since this is in the broker instance, this means that central is able to process only one message and the second one will never be written to the `r` channel.